### PR TITLE
Add timeout to the e2e command

### PR DESCRIPTION
**Pull request by abhinavmpandey08**
*Description of changes:*
Add timeout to the e2e command. This makes sure that the e2e test command itself times out instead of the SSM command timing out. With this change, we will be able to get the results of the tests that finished even if the instance timed out.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/ssm/command.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func RunCommand(session *session.Session, logger logr.Logger, instanceId, comman
 
 	logger.Info("Waiting for ssm command to finish")
 	var commandOut *ssm.GetCommandInvocationOutput
-	r := retrier.New(300*time.Minute, retrier.WithMaxRetries(2160, 60*time.Second))
+	r := retrier.New(310*time.Minute, retrier.WithMaxRetries(2160, 60*time.Second))
```
Comment (abhinavmpandey08) [inline internal/pkg/ssm/command.go:97]: Making SSM timeout slightly longer than e2e timeout so e2e timeout gets triggered before SSM timeout

File: internal/test/e2e/run.go [hunk 2]
```diff
@@ -205,7 +205,7 @@ func (e *E2ESession) runTests(regex string) (testCommandResult *testCommandResul
 	command := "GOVERSION=go1.16.6 gotestsum --junitfile=junit-testing.xml --raw-command --format=standard-verbose --hide-summary=all --ignore-non-json-output-lines -- test2json -t -p e2e ./bin/e2e.test -test.v"
 
 	if regex != "" {
-		command = fmt.Sprintf("%s -test.run \"%s\"", command, regex)
+		command = fmt.Sprintf("%s -test.run \"%s\" -test.timeout 300m", command, regex)
```
Comment (abhinavmpandey08) [inline internal/test/e2e/run.go:208]: This sets the e2e timeout to 300 minutes for each e2e test instance

File: internal/pkg/ssm/command.go (hunk 1 above)
Comment (g-gaston) [reply internal/pkg/ssm/command.go:97]: Can we make this configurable?
That way we can pass down the timeouts from the a central place.
If not, these two numbers need to be in sync in the two files, it's a hidden dep. Someone might go and change one of them not knowing about this dependency.

Comment (g-gaston) [inline internal/pkg/ssm/command.go:97]: I think we also need to update the `SendCommandInput`
I think we want `go test timeout` < `SSM command timeout` < `timeout in retrier that pulls the SSM command results`.
That guarantees the SSM command always gets the test results and that we always get the results of the SSM command.

**Review by g-gaston — COMMENTED** [review]
Great idea!

File: internal/test/e2e/run.go (hunk 2 above)
Comment (cxbrowne1207) [reply internal/test/e2e/run.go:208]: Does this cover the duration of the individual tests or all the tests being run? If it's the timeout for all the tests, should we make it 6 hours? 5.5 is pretty close to the normal time it's been taking to run a few of the provider test durations.

File: internal/pkg/ssm/command.go (hunk 1 above)
Comment (abhinavmpandey08) [reply internal/pkg/ssm/command.go]: done

Comment (abhinavmpandey08) [reply internal/pkg/ssm/command.go]: Is this what you meant? https://github.com/aws/eks-anywhere/pull/5642/files#diff-ec745d2a43e1da6e449129ed0106f64d40ac71e38690f7970f0d3d440ed44889R125-R130

File: internal/test/e2e/run.go [hunk 3]
```diff
@@ -217,6 +225,7 @@ func (e *E2ESession) runTests(regex string) (testCommandResult *testCommandResul
 		e.logger.V(4),
 		e.instanceId,
 		command,
+		defaultSSMTimeout,
```
Comment (g-gaston) [inline internal/test/e2e/run.go:228]: maybe we can do 
```suggestion
		e2eTimeout + e2eSSMTimeoutPadding,
```

and having that as a const (whichever value you decide)
```go
e2eSSMTimeoutPadding = 10 * Minute
```

File: internal/pkg/ssm/command.go [hunk 4]
```diff
@@ -94,7 +98,7 @@ func RunCommand(session *session.Session, logger logr.Logger, instanceId, comman
 
 	logger.Info("Waiting for ssm command to finish")
 	var commandOut *ssm.GetCommandInvocationOutput
-	r := retrier.New(300*time.Minute, retrier.WithMaxRetries(2160, 60*time.Second))
+	r := retrier.New(timeout, retrier.WithMaxRetries(2160, 60*time.Second))
```
Comment (g-gaston) [inline internal/pkg/ssm/command.go:101]: I would make this one slightly bigger
```suggestion
    // Making the retrier wait for longer than the provided SSM timeout to make sure
    // we always get the output results.
	r := retrier.New(timeout + 5 * time.Minute, retrier.WithMaxRetries(2160, 60*time.Second))
```

Now that I think about it, maybe we need to change the option as well

```suggestion
	// Making the retrier wait for longer than the provided SSM timeout to make sure
    // we always get the output results.
	r := retrier.New(timeout + 5 * time.Minute, retrier.WithMaxRetries(math.MaxInt, 60*time.Second))
```
That should guarantee "infinite" (not really) retries, so we only error out if max timeout is reached.

**Review by g-gaston — APPROVED** [review]

**Pull request merged**